app/services/SubmitMessageService.scala

```diff
@@ -79,6 +83,50 @@ class SubmitMessageService @Inject()(
       }
     }
 
+  def submitIe007Message(arrivalId: ArrivalId, messageId: MessageId, message: MovementMessageWithStatus, mrn: MovementReferenceNumber)(
```

Comment:
Are submitMessage and submitIe007Message doing the same thing?

Reply:
They are 😞
This was duplicated because this method is more generic, but I don't think that its the final design. The main limitation that made this a necessary, but temporary 🤞, duplication is the repository methods that we call. Those methods are highly specialized to the use cases and don't have a way to abstract over them. 

The Semigroup is an attempt to do that so that you have different types of updates that you give to the repository layer and you can combine them together and ask the repository layer to do its thing using the combined updates (or single update type).